### DISCUSSION TITLE
Commonmark & gfm parsing difference for links + raw HTML

### DISCUSSION THREAD

**Discussion by TripleCamera** [jgm/pandoc#9406] [Q&A] [answered]
Hi. I am using pandoc to convert markdown to html. For the following lines:
```markdown
aaabbb

aaa<span></span>bbb

[link](https://baidu.com)aaabbb

[link](https://baidu.com)aaa<span></span>bbb
```

When the source language is [`commonmark`](https://pandoc.org/try/?params=%7B%22text%22%3A%22aaabbb%5Cn%5Cnaaa%3Cspan%3E%3C%2Fspan%3Ebbb%5Cn%5Cn%5Blink%5D%28https%3A%2F%2Fbaidu.com%29aaabbb%5Cn%5Cn%5Blink%5D%28https%3A%2F%2Fbaidu.com%29aaa%3Cspan%3E%3C%2Fspan%3Ebbb%5Cn%22%2C%22to%22%3A%22html%22%2C%22from%22%3A%22commonmark%22%2C%22standalone%22%3Afalse%2C%22embed-resources%22%3Afalse%2C%22table-of-contents%22%3Afalse%2C%22number-sections%22%3Afalse%2C%22citeproc%22%3Afalse%2C%22html-math-method%22%3A%22plain%22%2C%22wrap%22%3A%22auto%22%2C%22highlight-style%22%3Anull%2C%22files%22%3A%7B%7D%2C%22template%22%3Anull%7D), the raw HTML tags are preserved when following a link:
```html
<p>aaabbb</p>
<p>aaa<span></span>bbb</p>
<p><a href="https://baidu.com">link</a>aaabbb</p>
<p><a href="https://baidu.com">link</a>aaa<span></span>bbb</p>
```

However, when the source language is [`gfm`](https://pandoc.org/try/?params=%7B%22text%22%3A%22aaabbb%5Cn%5Cnaaa%3Cspan%3E%3C%2Fspan%3Ebbb%5Cn%5Cn%5Blink%5D%28https%3A%2F%2Fbaidu.com%29aaabbb%5Cn%5Cn%5Blink%5D%28https%3A%2F%2Fbaidu.com%29aaa%3Cspan%3E%3C%2Fspan%3Ebbb%5Cn%22%2C%22to%22%3A%22html%22%2C%22from%22%3A%22gfm%22%2C%22standalone%22%3Afalse%2C%22embed-resources%22%3Afalse%2C%22table-of-contents%22%3Afalse%2C%22number-sections%22%3Afalse%2C%22citeproc%22%3Afalse%2C%22html-math-method%22%3A%22plain%22%2C%22wrap%22%3A%22auto%22%2C%22highlight-style%22%3Anull%2C%22files%22%3A%7B%7D%2C%22template%22%3Anull%7D), they are escaped:
```html
<p>aaabbb</p>
<p>aaa<span></span>bbb</p>
<p><a href="https://baidu.com">link</a>aaabbb</p>
<p><a
href="https://baidu.com">link</a>aaa&lt;span&gt;&lt;/span&gt;bbb</p>
```

I have read the specs and couldn't find any difference for links & raw HTML. Is this a bug in Pandoc?

**Comment by jgm** [comment]
Here's what I'm seeing (latest pandoc release):
```
% pandoc -f gfm
<p><a href="https://baidu.com">link</a>aaa<span></span>bbb</p>
<p><a href="https://baidu.com">link</a>aaa<span></span>bbb</p>
```
That looks fine!  What version of pandoc are you using?

**Reply by TripleCamera** [reply]
I am using the latest version, 3.1.11.1. I also tried [demos](https://pandoc.org/demos.html) on the official website.
```
C:\Users\EricQiu>pandoc -f gfm -t html
aaabbb

aaa<span></span>bbb

[link](https://baidu.com)aaabbb

[link](https://baidu.com)aaa<span></span>bbb
^Z
<p>aaabbb</p>
<p>aaa<span></span>bbb</p>
<p><a href="https://baidu.com">link</a>aaabbb</p>
<p><a
href="https://baidu.com">link</a>aaa&lt;span&gt;&lt;/span&gt;bbb</p>

C:\Users\EricQiu>pandoc -v
pandoc 3.1.11.1
Features: +server +lua
Scripting engine: Lua 5.4
User data directory: C:\Users\EricQiu\AppData\Roaming\pandoc
Copyright (C) 2006-2023 John MacFarlane. Web: https://pandoc.org
This is free software; see the source for copying conditions. There is no
warranty, not even for merchantability or fitness for a particular purpose.

```

**Reply by jgm** [reply]
Odd. This is what I get with the same input and pandoc version:
```
% pandoc
aaabbb

aaa<span></span>bbb

[link](https://baidu.com)aaabbb

[link](https://baidu.com)aaa<span></span>bbb
^D
<p>aaabbb</p>
<p>aaa<span></span>bbb</p>
<p><a href="https://baidu.com">link</a>aaabbb</p>
<p><a href="https://baidu.com">link</a>aaa<span></span>bbb</p>
```
However, I *can* reproduce it in [try pandoc](https://pandoc.org/try/?params=%7B%22text%22%3A%22aaabbb%5Cn%5Cnaaa%3Cspan%3E%3C%2Fspan%3Ebbb%5Cn%5Cn%5Blink%5D%28https%3A%2F%2Fbaidu.com%29aaabbb%5Cn%5Cn%5Blink%5D%28https%3A%2F%2Fbaidu.com%29aaa%3Cspan%3E%3C%2Fspan%3Ebbb%5Cn%5Cn%22%2C%22to%22%3A%22html5%22%2C%22from%22%3A%22gfm%22%2C%22standalone%22%3Afalse%2C%22embed-resources%22%3Afalse%2C%22table-of-contents%22%3Afalse%2C%22number-sections%22%3Afalse%2C%22citeproc%22%3Afalse%2C%22html-math-method%22%3A%22plain%22%2C%22wrap%22%3A%22auto%22%2C%22highlight-style%22%3Anull%2C%22files%22%3A%7B%7D%2C%22template%22%3Anull%7D)

My current guess is that this has to do with line endings. (Web forms and windows use CR+LF, whereas my setup is LF.)  I'll create a bug report based on this.

**Comment by jgm** [comment, marked as answer]
See ~~#9416~~ https://github.com/jgm/commonmark-hs/issues/147
